stdlib/public/Concurrency/PartialAsyncTask.swift

```diff
@@ -106,9 +114,76 @@ extension UnownedJob: CustomStringConvertible {
 
 #if !SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
 
+/// Deprecated equivalent of ``ExecutorJob``.
+///
+/// A unit of scheduleable work.
+///
+/// Unless you're implementing a scheduler,
+/// you don't generally interact with jobs directly.
 @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
 @available(*, deprecated, renamed: "ExecutorJob")
-public typealias Job = ExecutorJob
+@frozen
+@_moveOnly
+public struct Job: Sendable {
+  internal var context: Builtin.Job
+
+  @usableFromInline
+  internal init(context: __owned Builtin.Job) {
+    self.context = context
+  }
+
+  public init(_ job: UnownedJob) {
+    self.context = job._context
+  }
+
+  public init(_ job: __owned ExecutorJob) {
+    self.context = job.context
+  }
+
+  public var priority: JobPriority {
+    let raw = _swift_concurrency_jobPriority(UnownedJob(context: self.context))
+    return JobPriority(rawValue: raw)
+  }
+
+  // TODO: move only types cannot conform to protocols, so we can't conform to CustomStringConvertible;
+  //       we can still offer a description to be called explicitly though.
+  public var description: String {
+    let id = _getJobTaskId(UnownedJob(context: self.context))
+    /// Tasks are always assigned an unique ID, however some jobs may not have it set,
+    /// and it appearing as 0 for _different_ jobs may lead to misunderstanding it as
+    /// being "the same 0 id job", we specifically print 0 (id not set) as nil.
+    if (id > 0) {
+      return "\(Self.self)(id: \(id))"
+    } else {
+      return "\(Self.self)(id: nil)"
```

Comment:
```suggestion
      return "Job(id: \(id))"
    } else {
      return "Job(id: nil)"
```
or if you could cherry-pick this from main on top of this PR we can avoid colliding :) https://github.com/apple/swift/commit/8511317a9720dbfafd83901d60c7030834687243

Reply:
Will do, thanks!